client/core/trade.go

```diff
@@ -1584,6 +1584,21 @@ func (c *Core) swapMatchGroup(t *trackedTrade, matches []*matchTracker, errs *er
 		errs.add("not broadcasting swap while DEX %s connection is down (could be revoked)", t.dc.acct.host)
 		return
 	}
+
+	// Use a higher swap fee rate if a local estimate is higher than the
+	// prescribed rate, but not higher than the funded (max) rate.
+	var freshRate uint64
+	if r, ok := t.wallets.fromWallet.feeRater(); ok {
+		freshRate, _ = r.FeeRate()
+	}
+	if freshRate == 0 { // either not a FeeRater, or FeeRate failed
+		freshRate = t.dc.bestBookFeeSuggestion(fromAsset.ID)
+	}
+	if highestFeeRate < freshRate && freshRate <= t.metaData.MaxFeeRate {
+		c.log.Infof("Prescribed %v fee rate %v looks low, using %v",
+			fromAsset.Symbol, highestFeeRate, freshRate)
+		highestFeeRate = freshRate
+	}
```

Comment:
`else if freshRate > t.metaData.MaxFeeRate { highestRate = t.metaData.MaxFeeRate }` ?

Reply:
Good catch.  Reworked this logic.

What do you think about the tangential issue I called out in https://github.com/decred/dcrdex/pull/1501#discussion_r818178006?  It's only an issue for a non-`FeeRater` with no `orderbook` subscription.